src/main/java/cloud/cholewa/game/logic/GameEngine.java

```diff
@@ -0,0 +1,35 @@
+package cloud.cholewa.game.logic;
+
+import cloud.cholewa.exceptions.GameException;
+import cloud.cholewa.game.components.Player;
+
+import java.util.List;
+import java.util.Scanner;
+
+import static cloud.cholewa.common.ConsoleControl.showConsoleMessage;
+import static cloud.cholewa.game.messages.GameMessage.GAME_END;
+import static cloud.cholewa.game.messages.GameMessage.WELCOME_MESSAGE;
+
+public class GameEngine {
+
+    public static void run() {
+        Scanner scanner = new Scanner(System.in);
+
+        int numberOfPlayer;
+        List<Player> playerList;
+
+        showConsoleMessage("");
+        showConsoleMessage(WELCOME_MESSAGE, 1);
+
+        numberOfPlayer = GameStarter.setNumberOfPlayers(scanner);
+        playerList = (GameStarter.setPlayersName(scanner, numberOfPlayer));
+
+        //to remove in final version
+        showConsoleMessage("", 1);
+        System.out.println("Liczba graczy: " + numberOfPlayer);
+        playerList.forEach(System.out::println);
+
+        scanner.close();
+        throw new GameException(GAME_END);
```

Comment:
Ten throw jest niepotrzebny, w tym momencie run() naturalnie się skończy, wróci do App i zakończy działanie programu. No chyba, że koniecznie chcesz mieć informację o tym, że gra się zakończyła. Jeśli tak to łamiesz w tym momencie jedną z zasad programowaia [https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why](https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why) . Zamiast tego możesz po prostu napisać GAME_END w tym miejscu, albo w App, gdy GameEngine.run() zakończ wykonywanie, a najlepiej to bloku finally w try-cache, wyglądałoby to +- tak (łącząc z powyższym komentarzem)
```
try{ 
   GameEngine.run()
 } catch (IllegalStateException | InputMismatchEception | GameException e)  { 
  "handling expected exception" 
} catch (Exception e) {
  "handling unexpected exception" 
} finally {
   "Game Over" 
 }
```

Reply:
Kurcze nawet o tym nie pomyślałem. Dzięki.
Poprawione.